runhouse/servers/cluster_servlet.py

```diff
@@ -217,3 +244,141 @@ async def aclear_all_references_to_env_servlet_name(self, env_servlet_name: str)
         for key in deleted_keys:
             self._key_to_env_servlet_name.pop(key)
         return deleted_keys
+
+    ##############################################
+    # Cluster status functions
+    ##############################################
+
+    async def asend_status_info_to_den(self):
+        while True:
+            logger.info("Sending cluster status to Den")
+            try:
+                interval_size = (await self.aget_cluster_config()).get(
+                    "den_status_ping_interval"
+                )
+                if interval_size == -1:
+                    break
+                status: ResourceStatusData = await self.astatus()
+                status_data = {
+                    "status": "running",
+                    "resource_type": status.cluster_config.get("resource_type"),
+                    "data": dict(status),
+                }
+                cluster_uri = rns_client.format_rns_address(
+                    (await self.aget_cluster_config()).get("name")
+                )
+                api_server_url = status.cluster_config.get(
+                    "api_server_url", rns_client.api_server_url
+                )
+                post_status_data_resp = requests.post(
+                    f"{api_server_url}/resource/{cluster_uri}/cluster/status",
+                    data=json.dumps(status_data),
+                    headers=rns_client.request_headers(),
+                )
+                if post_status_data_resp.status_code != 200:
+                    logger.error(
+                        f"({post_status_data_resp.status_code}) Failed to send cluster status check to Den: {post_status_data_resp.text}"
+                    )
+                else:
+                    logger.info(
+                        f"Successfully updated cluster status in Den. Next status check will be in {round(interval_size / 60, 2)} minutes."
+                    )
+            except Exception as e:
+                logger.error(
+                    f"Cluster status check has failed: {e}. Please check cluster logs for more info."
+                )
+                logger.warning(
+                    f"Temporarily increasing the interval between two consecutive status checks. Next status check will be in {round(INCREASED_STATUS_CHECK_INTERVAL / 60, 2)} minutes. For changing the interval size, please restart the server."
```

Comment:
wouldn't changing the interval size also require updating the `den_status_ping_interval` on the client side as well? (in addition to restarting the server)

Reply:
When we restart the server, the` den_status_ping_interval` is updated in the cluster_config.json which is saved on the cluster. Later, the `send_status_info_to_den` job extracts the value from the cluster_config. I could add to the log something like "When restarting, please provide a new interval size. If not provided, interval size will be set to {DEFAULT_INTERVAL_SIZE}". @jlewitt1

Reply:
cool yea i meant just a way to indicate more specifically how to actually provide that new interval, makes sense